components/transactions/TransactionDetails.js

```diff
@@ -129,36 +146,36 @@ const TransactionDetails = ({ displayActions, transaction, isRoot, isHostAdmin,
           {displayActions && ( // Let us overide so we can hide buttons in the collective page
             <React.Fragment>
               <Flex justifyContent="flex-end">
-                {permissions?.canRefund && !isRefunded && (
+                {showRefundButton && (
                   <TransactionRefundButton
                     id={id}
                     disabled={userCanRejectOrRefund}
                     onMutationSuccess={onMutationSuccess}
                   />
                 )}
-                {permissions?.canReject && !isRejected && showRejectContribution && (
+                {showRejectButton && (
                   <TransactionRejectButton
                     id={id}
                     canRefund={permissions?.canRefund && !isRefunded}
                     disabled={userCanRejectOrRefund}
                     onMutationSuccess={onMutationSuccess}
                   />
                 )}
-              </Flex>
-              {!(permissions?.canRefund && !isRefunded) && // Just so we don't polute the UI, the Credit transaction will display the Download button
-                permissions?.canDownloadInvoice && (
+                {showDownloadInvoiceButton && (
                   <StyledButton
                     buttonSize="small"
                     loading={loadingInvoice}
                     onClick={downloadInvoiceWith({ transactionUuid: uuid, toCollectiveSlug: toAccount.slug })}
                     minWidth={140}
                     background="transparent"
                     textTransform="capitalize"
+                    ml={showRejectButton ? 2 : 0}
```

Comment:
You can simplify `ml` here and in `TransactionRejectButton` by removing the conditional and keeping `ml={2}` only in the last two buttons (since they're aligned to 'flex-end').